src/v/cloud_storage/cache_service.cc

```diff
@@ -1383,9 +1393,37 @@ cache::trim_carryover(uint64_t delete_bytes, uint64_t delete_objects) {
     co_return result;
 }
 
+void cache::maybe_background_trim() {
+    uint64_t target_bytes = uint64_t(
+      _max_bytes * _trim_threshold_bytes_percent().value_or(100.0) / 100.0);
+    uint32_t target_objects = uint32_t(
+      _max_objects() * _trim_threshold_objects_percent().value_or(100.0)
+      / 100.0);
```

Comment:
nit: maybe we can return early without any math if they're both null

Reply:
Done.